backend/internal/usecases/update_post_usecase.go

```diff
@@ -0,0 +1,37 @@
+package usecases
+
+import (
+	"errors"
+	"myapp/internal/controllers/schema"
+	"myapp/internal/entities"
+	"myapp/internal/exception"
+	"myapp/internal/interfaces"
+)
+
+type UpdatePostUsecase struct {
+	postRepository interfaces.PostRepository
+}
+
+func NewUpdatePostUsecase(
+	postRepository interfaces.PostRepository,
+) *UpdatePostUsecase {
+	return &UpdatePostUsecase{
+		postRepository: postRepository,
+	}
+}
+
+func (u *UpdatePostUsecase) Execute(request schema.PostRequest, userId int64, postId int64) (*entities.Post, error) {
+	post, err := u.postRepository.GetById(postId)
+	if post == nil || err == errors.New("post not found") {
+		return nil, exception.ErrPostNotFound
+	} else if err != nil {
+		return nil, err
+	}
```

Comment:
GetById は post 見つからん時に nil, nil で返るから、こんな感じのエラーハンドリングなりそう
```suggestion
	post, err := u.postRepository.GetById(postId)
	if err != nil {
		return nil, err
	} else if post == nil {
	        return nil, exception.ErrPostNotFound
	}
```

Reply:
修正しました！ [265f281](https://github.com/givery-bootcamp/dena-2024-team10/pull/83/commits/265f2814fec9d3ad0647d9365181d7515a03ec0c)
ここ、本来は昨日Zoomで話した、「repositoryのpost not foundをusecaseでErrorとして返したい」みたいに言ってた所に対応したかったので適切な場所へ書き直しました！